BEST_PRACTICES.md

```diff
@@ -0,0 +1,81 @@
+# Introduction
+This is a collection of tips, advice, gotchas and other best practices
+for using the [Jenkins Workflow
+plugin](https://github.com/jenkinsci/workflow-plugin/blob/master/README.md). Contributions
+and comments are happily accepted.
+
+# General tips and advice
+* Do everything that makes sense there within `stage`s. This makes your builds
+make more sense to visualize, etc.
+* Do all real work that involves shelling out, building, etc, within
+`node` blocks, so that it actually happens on an executor.
```

Comment:
Perhaps append "instead of the master node" onto the end of this. It's otherwise not apparently to a new-comer why this might be important